### PR TITLE
[ISSUE-15] Code Formatter Support

### PR DESCRIPTION
**Pull request by belloibrahv**
**Description:**

**Background:**
- During the review of code formatting tools for our project, I assessed `Autopep8` and `Black` to determine the most suitable choice based on our core requirements.

**Findings:**
- `Autopep8` successfully meets our essential criteria regarding single quotes, line length, and configuration options. However, it lacks a comprehensive plugin system.
- `Black`, on the other hand, not only addresses our plugin and configurability needs but also provides a robust and extensible solution. Despite its stricter style, it offers the flexibility to accommodate our preferences.

**Decision:**
- After careful consideration, I propose adopting `Black` as our project's code formatter due to its superior customizability and extensibility.

Implementation Steps:
1. Installed Black using the command: `pipenv install black`.
2. Added the required configuration keys to the `project.toml file`
3. Executed the formatting command: `black src tests` to apply the formatting changes.

**Note**
- Review the changes made by `Black` to ensure alignment with project coding standards.
- Test the formatted code to verify compatibility with our existing codebase.

### CODE REVIEW

**Comment by belloibrahv** [comment]
Certainly, enable single quotes in Black by adding `skip-string-normalization = true` to your `pyproject.toml` file. Check the file to see the applied modification.

File: pyproject.toml [hunk 1]
```diff
@@ -22,3 +22,7 @@ classifiers = [
 Homepage  = "https://github.com/aidubtech/mezages"
 Issues    = "https://github.com/aidubtech/mezages/issues"
 Changelog = "https://github.com/aidubtech/mezages/blob/main/CHANGELOG.md"
+
+[tool.black]
```
Comment (abdulfataiaka) [inline pyproject.toml:26]: Lets move this to the top of the file

**Comment by abdulfataiaka** [comment]
Have you run black against the codebase? Did you ensure that flake8 does not report any issue after running black agianst the codebase?

**Comment by belloibrahv** [comment]
When I executed Black with the command `flake8 src tests`, it detected an issue in `src/mezages/subjects.py` at line 49:48. Specifically, it highlighted 'E203 whitespace before `:`.' The issue arises when formatting the code, changing it from `child_sub_path = child_path[len(f'{path}.'):]` to `child_sub_path = child_path[len(f'{path}.') :]`. Unfortunately, there is no direct way to prevent Black from introducing whitespace in this particular scenario.

To avoid conflicts and ensure compatibility with Flake8, the recommended approach is to add `# ffg: skip` as a comment at the end of the line. This instructs Black to skip formatting for that specific line of code, preventing potential issues reported by Flake8 due to formatting changes introduced by Black.

**Review by abdulfataiaka — APPROVED** [review]
👍

**Review by yuusuf4real — APPROVED** [review]
well done

**Pull request merged**